maestro-client/src/main/java/maestro/drivers/IOSDriver.kt

```diff
@@ -187,6 +189,41 @@ class IOSDriver(
         }
     }
 
+    fun viewHierarchy(): TreeNode {
+        val hierarchy = iosDevice.viewHierarchy().get() ?: return TreeNode()
+        return mapViewHierarchy(hierarchy)
+    }
+
+    private fun mapViewHierarchy(element: AXElement): TreeNode {
```

Comment:
CMIIW driver is responsible to convert the native Hierarchy to cross-platform `TreeNode` are you suggesting we should have a separate abstraction that should take responsibility for that conversion for both Android and iOS drivers? Heads up, this was already there for `subTree` call so we might need to also consider refactoring that

Reply:
for me drivers already have too many responsibilities (i.e. IOSDriver is >500 lines already), so it is better to abstract some of this logic. sounds like it is out of scope from this PR though.

Reply:
I doubted to put this logic in AXElement, but AXElement is in maestro-ios, and therefore can't access TreeNode which is in maestro-client. Then I considered adding this method in TreeNode, which would be messy (why should TreeNode know about AXElement). 

I suggest to keep the logic where it is now.

Lets look at this later, but fwiw, I think we can extract TreeNode into a helper module and have maestro-ios use it directly. That way we don't have to do all the translations between the different models. It would also be easier to add eg. the frame bounds and centerpoint as calculated model values.